libvast/vast/experimental_table_slice.hpp

```diff
@@ -97,6 +101,12 @@ class experimental_table_slice final {
   [[nodiscard]] std::shared_ptr<arrow::RecordBatch>
   record_batch() const noexcept;
 
+  /// Provides a data column for an offset.
+  /// @param column The column index
+  /// @returns The arrow array backing the (flattened) column
+  [[nodiscard]] std::shared_ptr<arrow::Array>
+  column_array(table_slice::size_type column) const noexcept;
```

Comment:
Should this be part of the public API? It's not (yet) used by table_slice directly, right?

Reply:
After our discussion about adding attributes to the state, I realized that amending the API does work here, but is not going to work when merging this table slice implementation back into `arrow_table_slice<...::experimental>`. I fixed it in that other branch, but leaving it here as the file is not going to be around for long.